target/linux/brcmiproc/bcm58800/base-files/lib/upgrade/platform.sh

```diff
@@ -0,0 +1,17 @@
+platform_check_image() {
+  # i know no way to verify the image
+  return 0;
+}
+
+platform_do_upgrade() {
+  sync
+  get_image "$1" | dd of=/dev/mmcblk0p5 bs=2M conv=fsync
+  sleep 1
```

Comment:
why the sync before, shouldn't you do that after writing it? And where do you copy the kernel?

Reply:
Have removed this file for now. Looks like having it is not mandatory. The samsung target and some of the other targets don't have it. May add a more compete update script later that updates both kernel and rootfs and also uses recovery pertitions etc.